cleverhans/attacks.py

```diff
@@ -514,6 +514,115 @@ def parse_params(self, eps=2.0, num_iterations=1, xi=1e-6, clip_min=None,
         return True
 
 
+class CarliniWagnerL2(Attack):
```

Comment:
Yeah, there are different interfaces (e.g., there's no binary search in l0/li), but those have other constants l2 doesn't.